src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritQueryHandlerHttp.java

```diff
@@ -0,0 +1,497 @@
+   * @throws GerritQueryException if Gerrit reports an error with the query
+   * @throws IOException  for some other IO problem.
+   */
+  public List<String> queryJson(String queryString, boolean getPatchSets, boolean getCurrentPatchSet,
+                                boolean getFiles, boolean getCommitMessage)
+      throws GerritQueryException, IOException {
+
+    List<String> list = new ArrayList<>();
+
+    runQuery(queryString, getPatchSets, getCurrentPatchSet, getFiles, getCommitMessage, false,
+        new GerritQueryHandlerHttp.LineVisitor() {
+          @Override
+          public void visit(String line) {
+            list.add(line);
+          }
+        });
+    return list;
+  }
+
+  //CS IGNORE RedundantThrows FOR NEXT 24 LINES. REASON: Informative.
+  //CS IGNORE JavadocMethod FOR NEXT 20 LINES. REASON: It is there.
+
+  /**
+   * Runs the query on the Gerrit server and lets the provided visitor handle each line in the result.
+   *
+   * @param queryString        the query. (e.g "limit=1")
+   * @param getPatchSets       if all patch-sets of the projects found should be included in the result.
+   *                           Meaning if --patch-sets should be appended to the command call.
+   * @param getCurrentPatchSet if the current patch-set for the projects found should be included in the result.
+   *                           Meaning if --current-patch-set should be appended to the command call.
+   * @param getFiles           if changed files list should be included in the result.
+   *                           Meaning if --files should be appended to the command call.
+   * @param getCommitMessage   if full commit message should be included in the result.
+   *                           Meaning if --commit-message should be appended to the command call.
+   * @param getComments        if patchset comments should be included in the results.
+   *                           Meaning if --comments should be appended to the command call.
+   *                           //@param visitor the visitor to handle each line in the result.
+   * @throws GerritQueryException if a visitor finds that Gerrit reported an error with the query.
+   * @throws IOException          for some other IO problem.
+   */
+  private void runQuery(String queryString, boolean getPatchSets, boolean getCurrentPatchSet, boolean getFiles,
+                            boolean getCommitMessage, boolean getComments, LineVisitor visitor)
+      throws GerritQueryException, IOException {
+
+
+    StringBuilder str = new StringBuilder(httpBaseUrl);
+
+    str.append("/a/changes/?q=").append(queryString);
+
+    // map the cl-arguments from: https://gerrit-review.googlesource.com/Documentation/cmd-query.html
+    // to Rest-API-arguments:          https://gerrit-review.googlesource.com/Documentation/rest-api-changes.html
+
+    if (getPatchSets) {
+      //ALL_REVISIONS: describe all revisions, not just current.
+      str.append("&o=ALL_REVISIONS");
```

Comment:
macht es ggf. Sinn, diese URL-Parameter ("&o=ALL_REVISIONS") in Konstanten auszulagern? Dann könnte man diese auch u.A. in den Tests verwenden